keps/sig-auth/1205-bound-service-account-tokens/README.md

```diff
@@ -511,3 +586,54 @@ are properly reloading tokens by:
 - **Will enabling / using this feature result in non-negligible increase of
   resource usage (CPU, RAM, disk, IO, ...) in any components?** it adds a
   token minting operation in the API server every ~48 minutes for every pod.
+
+### Troubleshooting
+
+The Troubleshooting section currently serves the `Playbook` role. We may
+consider splitting it into a dedicated `Playbook` document (potentially with
+some monitoring details). For now, we leave it here.
+
+- **How does this feature react if the API server and/or etcd is
+  unavailable?**
+
+  - TokenRequest API is unavailable
+  - configmap containing API server CA bundle cannot be created or fetched
+
+* **What are other known failure modes?**
+
+  - failure to issue token via token subresource
+
+    - Detection: check `apiserver_request_total` with labels
+      `group="",version="v2",resource="serviceaccounts",subresource="token"`
+    - Mitigations: resort to legacy secret based token.
+    - Diagnostics: "failed to generate token" in kube-apiserver log.
+    - Testing: [e2e test](https://k8s-testgrid.appspot.com/sig-auth-gce#gce&width=5&include-filter-by-regex=ServiceAccounts%20should%20mount%20projected%20service%20account%20token)
+
+  - failure to create root CA config map
+
+    - Detection: check kube-controller-manager log
+    - Mitigations: resort to legacy secret based token.
+    - Diagnostics: "syncing [namespace]/[configmap name] failed" in
+      kube-controller-manager log.
+    - Testing: [e2e test](https://k8s-testgrid.appspot.com/sig-auth-gce#gce&width=5&include-filter-by-regex=ServiceAccounts%20should%20guarantee%20kube-root-ca.crt%20exist%20in%20any%20namespace)
+
+  - kubelet fails to renew token
+
+    - Detection: check kubelet log
```

Comment:
does the kubelet emit any metrics when it cannot obtain or renew a bound token? should it?

Reply:
kubelet simply calls TokenRequest API if it doesn't find one in the cache. i think metrics for TokenRequest can provide us with a good visibility.